### ISSUE TITLE
upload_subfolder returns a folder for each character

### ISSUE THREAD

**Issue by gregoriopellegrino** [edoburu/django-private-storage#11]
Hi,
I'm making a simple function from defining an upload subfolder:

```python
def subfolder(instance):
    return os.path.join("statements", str(instance.ID))
```

The path where the file is uploaded is: `s/t/a/t/e/m/e/n/t/s/2/my_file.pdf `

How can I fix it?

**Comment by gregoriopellegrino** [comment]
Found! You have to change line #67 in file `fields.py` with:
```
subdirs = [self.storage.get_valid_name(dir) for dir in os.path.split(subdir_func(instance))]
```

Gregorio

**Comment by thecardcheat** [comment]
Good find! However, `os.path.split()` is always going to return `(head, tail)` so calling `get_valid_name()` on `head` will strip the `/` out of the path. I think it should be alright with how I have done it in #12.

**Comment by gregoriopellegrino** [comment]
Perfect, thanks!

**Comment by thecardcheat** [comment]
@gregoriopellegrino The issue isn't resolved until the PR is accepted so I think you want to keep this open.

**Comment by vdboor** [comment]
Fixed in v2.0!
